ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala

```diff
@@ -134,17 +134,7 @@ private[kvutils] object InputsAndEffects {
               .fold(effects.updatedContractKeys)(keyWithMaintainers =>
                 effects.updatedContractKeys.updated(
                   globalKeyToStateKey(
-                    GlobalKey
-                      .build(
-                        create.coinst.template,
-                        keyWithMaintainers.key,
-                      )
-                      .fold(
-                        _ =>
-                          throw Err
-                            .InvalidSubmission("Contract IDs are not supported in contract keys."),
-                        identity,
-                      )
+                    GlobalKey.assertBuild(create.coinst.template, keyWithMaintainers.key)
```

Comment:
This looks like a change. You no longer throw `InvalidSubmission`, now we throw `IllegalArgumentException`. Does that matter @fabiotudone-da?

Reply:
No idea, it might, will check ASAP.

Reply:
Yes, I think it matters. We don't really treat `Err` errors differently from other exceptions, but they do have more meaningful messages. At the least we should probably wrap this in an `Err.InvalidSubmission`.

Reply:
(Same applies to all other exception type changes regarding `Err`.)